### PR TITLE
Check output tool and install Xcpretty if it is missing

### PR DESCRIPTION
**Pull request by BirmacherAkos**

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -91,7 +91,7 @@ workflows:
     - certificate-and-profile-installer:
     - path::./:
         inputs:
-        - output_tool: xcodebuild
+        - output_tool: $OUTPUT_TOOL
```
Comment (godrei) [inline bitrise.yml]: OUTPUT_TOOL is set to xcpretty (which is the default value for this input) please wire in test for xcodebuild output tool

File: main.go [hunk 2]
```diff
@@ -268,18 +268,40 @@ func main() {
...
+				log.Printf("Installing xcpretty")
+
+				if err = utils.InstallXcpretty(); err != nil {
+					log.Warnf("Failed to install xcpretty, error: %s", err)
+					log.Printf("Switching to xcodebuild for output tool")
+					outputTool = "xcodebuild"
+				}
+			}
+
+			if err == nil {
```
Comment (godrei) [inline main.go]: why is it needed?

File: main.go [hunk 3]
```diff
@@ -268,18 +268,40 @@ func main() {
...
+				xcprettyVersion, err := utils.XcprettyVersion()
+				if err != nil {
+					log.Warnf("Failed to determin xcpretty version, error: %s", err)
+					log.Printf("Switching to xcodebuild for output tool")
+					outputTool = "xcodebuild"
+				}
+				log.Printf("- xcprettyVersion: %s", xcprettyVersion.String())
```
Comment (godrei) [inline main.go]: do this after you made sure that xcpretty is available or switched to xcodebuild (if outputtool == xcpretty {...)

File: step.yml [hunk 4]
```diff
@@ -238,7 +238,7 @@ inputs:
       - xcpretty
       - xcodebuild
       is_required: true
-      is_expand: false
+      is_expand: true
```
Comment (godrei) [inline step.yml]: this is the default value, pls remove

File: utils/utils.go [hunk 5]
```diff
@@ -46,15 +48,31 @@ func XcodeBuildVersion() (XcodebuildVersionModel, error) {
 }
 
 // IsToolInstalled ...
-func IsToolInstalled(name string) bool {
```
Comment (godrei) [inline utils/utils.go]: was it only used for xcpretty?

File: utils/utils.go [hunk 6]
```diff
@@ -46,15 +48,31 @@ func XcodeBuildVersion() (XcodebuildVersionModel, error) {
...
-	cmd := command.New("which", name)
-	out, err := cmd.RunAndReturnTrimmedCombinedOutput()
-	return err == nil && out != ""
+func IsToolInstalled(name, version string) (bool, error) {
```
Comment (godrei) [inline utils/utils.go]: do we need this at all? it just wraps a single line of command

File: utils/utils.go [hunk 7]
```diff
@@ -46,15 +48,31 @@ func XcodeBuildVersion() (XcodebuildVersionModel, error) {
...
-	return IsToolInstalled("xcpretty")
+func IsXcprettyInstalled() (bool, error) {
+	return IsToolInstalled("xcpretty", "")
+}
+
+// InstallXcpretty ...
+func InstallXcpretty() error {
+	cmds, err := rubycommand.GemInstall("xcpretty", "")
+	if err != nil {
+		return fmt.Errorf("failed to create command model, error: %s", err)
```
Comment (godrei) [inline utils/utils.go]: failed to create %s install command, xcpretty

File: utils/utils.go [hunk 8]
```diff
@@ -46,15 +48,31 @@ func XcodeBuildVersion() (XcodebuildVersionModel, error) {
...
+	}
+
+	for _, cmd := range cmds {
+		log.Donef("$ %s", cmd.PrintableCommandArgs())
```
Comment (godrei) [inline utils/utils.go]: do not use green coloring, this is just a small utility, but not a big milestone in the step, like other Done logs

File: bitrise.yml (hunk 1 above)
Comment (BirmacherAkos) [reply bitrise.yml]: The test_objc workflow using the `xcodebuild` as an output tool.

**Pull request merged**